src/sys/bout_types.cxx

```diff
@@ -13,6 +14,17 @@ const std::string& safeAt(const std::map<T, std::string>& mymap, T t) {
   return found->second;
 }
 
+template <typename T>
+const T& safeAtReverse(const std::map<std::string, T>& mymap, std::string s) {
```

Comment:
You can drop the "reverse" -- it makes it sound like it's going to do something backwards! It's just an overload of `safeAt`